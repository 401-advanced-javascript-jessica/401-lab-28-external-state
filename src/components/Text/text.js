import React from 'react';

class Text extends React.Component {
  render() {
    return (
            <div>
                <div>Name: {this.props.name} </div>
                <p># of Updates: {this.props.count} </p>
            </div>
    );
  }
}

export default Text;
