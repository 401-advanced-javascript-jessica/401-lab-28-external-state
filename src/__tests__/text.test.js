import React from 'react';
import renderer from 'react-test-renderer';
import Text from '../components/Text/text.js';

describe("<Text />", () => {
    it('Text is rendered at application start', () => {
        let app = shallow(<Text />);
        expect(app.find('div').exists()).toBe(true);
    });

    it('renders properly', ()=> {
        const tree = renderer.create(<Text />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
