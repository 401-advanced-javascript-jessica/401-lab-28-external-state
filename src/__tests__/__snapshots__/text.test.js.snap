// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> renders properly 1`] = `
<div>
  <div>
    Name: 
     
  </div>
  <p>
    # of Updates: 
     
  </p>
</div>
`;
